### ISSUE TITLE
Inquiry About Missing test.py File

### ISSUE THREAD

**Issue by Zebrabeast** [HotChieh/BDRNet#3]
Dear HotChieh,
I hope you're doing well. I’m currently working with your BDRNet repository, but I noticed that the test.py file seems to be missing. Could you please let me know if there’s another way to test the model, or if I might have missed it?
Thank you for your help!

**Comment by HotChieh** [comment]
Thank you for your interest in our work. In some datasets, there is no explicit division into training, validation, and test sets.  Hence, the result on the validation set can be regarded as test set results. In our code, you can obtain validation set results using the function ```validate_v1()``` in [trainer.py](https://github.com/HotChieh/BDRNet/blob/main/trainer.py#L167). Therefore, there is no need to write a separate ```test.py```. You only need to: 

1. Set the [RESUME](https://github.com/HotChieh/BDRNet/blob/main/config.py#L24) status to ```True``` and specify the ```.pth``` [path](https://github.com/HotChieh/BDRNet/blob/main/config.py#L25). 
2. Add ```self.validate_V1()``` before [self.train()](https://github.com/HotChieh/BDRNet/blob/main/trainer.py#L86). 
 
After completing these steps, you can view the test results in either debug mode or normal mode.

If the dataset has a separate test set, simply modify the validation set path in [loading_data.py](https://github.com/HotChieh/BDRNet/blob/main/datasets/PUCPR/loading_data.py#L115) and repeat the above steps. You can refer to the PUCPR dataset as an example.

**Comment by Zebrabeast** [comment]
Dear HotChieh,
Thank you for your help with my previous question. Your response was very useful!
However, I’ve encountered a new issue during training. Specifically:
I am training the model on a dataset with around 14,000 images.
Regardless of whether I set the learning rate to 1e-3 or 1e-4, the model consistently achieves the best results (in terms of MAE and MSE) within the first 5 epochs.
lr:1e-4
<img width="626" alt="86d0509bfab3ee764411b76422cf755" src="https://github.com/user-attachments/assets/21124ee1-fee2-4d4c-8027-e9db5b4acb75" />
<img width="581" alt="06b00cf7305cff2f2019fcd3cf38792" src="https://github.com/user-attachments/assets/96fe302e-720e-466b-ace1-af47aacb39ab" />
<img width="515" alt="064fbd582d48775f3bbfbb358f6a2a5" src="https://github.com/user-attachments/assets/eb94c80a-357a-403d-943e-592faca92bcf" />
<img width="523" alt="b9d9b1833b54d015b0b456a27e983c4" src="https://github.com/user-attachments/assets/113c41f2-cbb4-4dde-880f-1e68ccfd8567" />
<img width="560" alt="53754c13d7f156bdd0ec17a15131aa9" src="https://github.com/user-attachments/assets/eb1198ea-56e9-4a0f-a0df-5a8022062c58" />
<img width="644" alt="08324c87313b8d44be98a767cfa114e" src="https://github.com/user-attachments/assets/3e305426-0b9e-4b6d-a1c3-6145135c671d" />
After that, both MAE and MSE start to increase as training continues.
Is this behavior normal? Could you please provide some guidance or suggestions to address this?
Thank you in advance for your help!

**Comment by HotChieh** [comment]
In my experience, maybe the model is overfitted due to the large amount of training data and the small difference in data distribution. That is, the distribution of the data set is too similar, which makes the model easy to fit, and the training data is large enough, so the model has achieved good fitting ability in the first few epochs. In order to find the exact root cause of the problem, I have the following suggestions:

1. Replace the training model with a simpler model, such as mcnn or basic VGG16BN, to see if the same problem occurs, and whether the best effect achieved is better than the current overfitting effect.

2. Switch the training set and test set to see if the same problem still occurs.

3. If it is determined that the model is overfitted, it can be handled from the perspective of solving overfitting. For example, add dataset enhancement methods (add noise, rotation, etc.) or apply dropout layers before the output layer of the model.

4. Replace the Adam optimizer with the SGD optimizer, because the SGD optimizer often takes longer to converge, and this gradual convergence process may prevent the model from falling into a local optimal solution.
